linux/mcompile/keymap/deadkey.cpp

```diff
@@ -1,468 +1,284 @@
 
   create_DKTable(dk_ComposeTable);
 
-	for( int i=0; i < (int) dk_ComposeTable.size()-1; i++) {
-		DeadKey *dk2 = new DeadKey(dk_ComposeTable[i][0]);
-		for ( int j=0; j< (int) dk_ComposeTable.size();j++) {
-			if(( dk_ComposeTable[i][0] == dk_ComposeTable[j][0]) && (IsKeymanUsedChar(dk_ComposeTable[j][1])))
-				dk2->KMX_AddDeadKeyRow(dk_ComposeTable[j][1],dk_ComposeTable[j][2]);
-		}
-		alDead.push_back(dk2);
-	}
-	return alDead;
+  for (int i = 0; i < (int)dk_ComposeTable.size() - 1; i++) {
+    DeadKey *dk2 = new DeadKey(dk_ComposeTable[i][0]);
+    for (int j = 0; j < (int)dk_ComposeTable.size(); j++) {
+      if ((dk_ComposeTable[i][0] == dk_ComposeTable[j][0]) && (IsKeymanUsedChar(dk_ComposeTable[j][1])))
+        dk2->KMX_AddDeadKeyRow(dk_ComposeTable[j][1], dk_ComposeTable[j][2]);
+    }
+    alDead.push_back(dk2);
+  }
+  return alDead;
 }
 
-void refine_alDead(KMX_WCHAR dk, std::vector<DeadKey*> &dkVec, std::vector<DeadKey*> *p_All_Vec) {
-	if (dk == 0)
-		return;
-
-	for (int j=0; j < (int) (*p_All_Vec).size()-1;j++) {
-		if (dk == (*p_All_Vec)[j]->KMX_GetDeadCharacter()) {
-			if(! found_dk_inVector(dk, dkVec)) {
-				dkVec.push_back((*p_All_Vec)[j]);
-				return;
-			}
-			else return;
-		}
-	}
+void refine_alDead(KMX_WCHAR dk, std::vector<DeadKey *> &dkVec, std::vector<DeadKey *> &r_All_Vec) {
+  if (dk == 0)
+    return;
+
+  for (int j = 0; j < (int)r_All_Vec.size() - 1; j++) {
+    if (dk == (r_All_Vec)[j]->KMX_GetDeadCharacter()) {
+      if (!found_dk_inVector(dk, dkVec)) {
+        dkVec.push_back((r_All_Vec)[j]);
+        return;
+      } else
+        return;
+    }
+  }
 }
 
-bool found_dk_inVector(KMX_WCHAR dk, std::vector<DeadKey*> &dkVec) {
-	int i=0;
-	if (dkVec.size() > 0) {
-		do {
-			if (dk == dkVec[i]->KMX_GetDeadCharacter())
-				return true;
-			i++;
-		} while (i < (int) dkVec.size());
-	}
-	return false;
+bool found_dk_inVector(KMX_WCHAR dk, std::vector<DeadKey *> &dkVec) {
+  for (int i = 0; i < dkVec.size(); i++) {
+    if (dk == dkVec[i]->KMX_GetDeadCharacter())
+      return true;
+  }
+  return false;
 }
 
-bool query_dk_combinations_for_specific_dk(vec_dword_2D * p_dk_ComposeTable, vec_dword_2D  &dk_SingleTable, KMX_DWORD dk) {
-	vec_dword_1D line;
-
-	for ( int i =0; i< (int) (*p_dk_ComposeTable).size(); i++) {
-		if (((*p_dk_ComposeTable)[i][0] == dk) && (IsKeymanUsedChar((*p_dk_ComposeTable)[i][1]))) {
-			line.push_back((*p_dk_ComposeTable)[i][0]);
-			line.push_back((*p_dk_ComposeTable)[i][1]);
-			line.push_back((*p_dk_ComposeTable)[i][2]);
-			dk_SingleTable.push_back(line);
-			line.clear();
-		}
-	}
+bool query_dk_combinations_for_specific_dk(vec_dword_2D &p_dk_ComposeTable, KMX_DWORD dk, vec_dword_2D &dk_SingleTable) {
+  vec_dword_1D line;
 
-	if (dk_SingleTable.size()>0)
-		return true;
-	else
-		return false;
+  for (int i = 0; i < (int)(p_dk_ComposeTable).size(); i++) {
+    if (((p_dk_ComposeTable)[i][0] == dk) && (IsKeymanUsedChar((p_dk_ComposeTable)[i][1]))) {
```

Comment:
we can save some parens here:

```suggestion
    if ((p_dk_ComposeTable[i][0] == dk) && IsKeymanUsedChar(p_dk_ComposeTable[i][1])) {
```
or even

```suggestion
    if (p_dk_ComposeTable[i][0] == dk && IsKeymanUsedChar(p_dk_ComposeTable[i][1])) {
```